### PR TITLE
Introduced PullRequestSyncService and refactored SyncCommand to use service

### PR DESCRIPTION
**Pull request by dennisdegreef**
Behaviour should be extracted away from the hexagonal architecture edge (namely the SyncCommand) towards a service which is responsible for this behaviour

### CODE REVIEW

**Comment by dennisdegreef** [comment]
@LonelyPullRequests

File: src/Infrastructure/Persistence/InMemoryPullRequestsRepository.php [hunk 1]
```diff
@@ -30,12 +31,33 @@ public function add(PullRequest $pullRequest)
...
+
+    /**
+     * @param PullRequest $pullRequest
+     *
+     * @return boolean
+     */
+    public function remove(PullRequest $pullRequest)
+    {
+        $this->pullRequests = $this->pullRequests->remove($pullRequest);
+        return true;
```
Comment (WouterSioen) [inline src/Infrastructure/Persistence/InMemoryPullRequestsRepository.php]: if you use PSR-2, there should be an empty line before a return statement.

**Comment by WouterSioen** [comment]
Looks good IMO

File: src/Infrastructure/Persistence/InMemoryPullRequestsRepository.php (hunk 1 above)
Comment (dennisdegreef) [inline src/Infrastructure/Persistence/InMemoryPullRequestsRepository.php]: Fixed in `568f31d`

**Comment by dennisdegreef** [comment]
@LonelyPullRequests Please help me get quality feedback from the community on this pull request.

**Pull request merged**